File: FurnitureStore/FurnitureStore.Models/Order/Order.cs

using FurnitureStore.Models.Enums;

namespace FurnitureStore.Models.Order
{
    public class Order
    {
        public long Id { get; set; }
        public DateTime OrderDate { get; set; }
        public Delivery Delivery { get; set; }
        public decimal TotalPrice { get; set; }
        public bool IsApproved { get; set; }
        public string CustomerId { get; set; }
        public User.User Customer { get; set; }
        public List<OrderItem.OrderItem> OrderItems { get; set; } = new();

    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/order/order.dart
import 'package:furniturestore_admin/models/order_item/order_item.dart';
import 'package:json_annotation/json_annotation.dart';

part 'order.g.dart';

@JsonSerializable()
class OrderModel {
  int? id;
  DateTime? orderDate;
  int? delivery;
  double? totalPrice;
  bool? isApproved;
  String? customerId;
  List<OrderItemModel>? orderItems;

  OrderModel({
    this.id,
    this.orderDate,
    this.delivery,
    this.totalPrice,
    this.isApproved,
    this.customerId,
    this.orderItems,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) =>
      _$OrderModelFromJson(json);

  Map<String, dynamic> toJson() => _$OrderModelToJson(this);
}
